File: Algorithms/094. Binary Tree Inorder Traversal/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
    	return inorder(root,new ArrayList<Integer>());
    }
    public List<Integer> inorder(TreeNode root,List<Integer> list){
    	if (root!=null) {
			list=inorder(root.left,list);
		    list.add(root.val);
			list=inorder(root.right,list);
		}
    	return list;
    }
}

File: Algorithms/094. Binary Tree Inorder Traversal/Solution.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val)
#         @val = val
#         @left, @right = nil, nil
#     end
# end

# @param {TreeNode} root
# @return {Integer[]}
def inorder_traversal(root)
    return root ? inorder_traversal(root.left)+[root.val]+inorder_traversal(root.right) : []
end